psutil/arch/solaris/process_as_utils.c

```diff
@@ -0,0 +1,302 @@
+/*
```

Comment:
If all these utilities are about the process environ() then this file should be called `environ.c`. I suggested the 2 different names (`environ.c` and `utils.c`) because in the previous PR where you addressed both environ() and cmdline() it seemed that some of these functions were shared by both environ and cmdline.

As a general rule, if a function is reused by multiple functionalities, then it's better to put them in a generic module, like `utils.c` or something. Anyway, in this case it seems like this should be called `environ.c`. We can move stuff around later.